symphony-bdk-core/src/main/java/com/symphony/bdk/core/client/loadbalancing/LoadBalancingStrategy.java

```diff
@@ -0,0 +1,44 @@
+package com.symphony.bdk.core.client.loadbalancing;
+
+import com.symphony.bdk.core.client.ApiClientFactory;
+import com.symphony.bdk.core.config.model.BdkConfig;
+import com.symphony.bdk.core.config.model.BdkLoadBalancingConfig;
+import com.symphony.bdk.core.config.model.BdkServerConfig;
+import com.symphony.bdk.gen.api.SignalsApi;
+
+import org.apiguardian.api.API;
+
+import java.util.List;
+
+/**
+ * Interface to provide a load balancing strategy, i.e. provide a new base URL when needed.
+ */
+@API(status = API.Status.INTERNAL)
+public interface LoadBalancingStrategy {
+
+  String getNewBasePath();
+
+  /**
+   * Returns a concrete implementation instance based on the provided inputs.
+   *
+   * @param config           the bdk configuration
+   * @param apiClientFactory the api client factory needed for the
+   *                         {@link com.symphony.bdk.core.config.model.BdkLoadBalancingMode#EXTERNAL} mode.
+   * @return a fully initialized instance whose implementation depends on the provided {@link BdkLoadBalancingConfig}
+   */
+  static LoadBalancingStrategy getInstance(BdkConfig config,
```

Comment:
Having a interface that depends on its implementations kind of defeats its purpose

I would move that to a LoadBalancingStrategyFactory (wooh that is a long name....)